### PR TITLE
Add prune & event blacklist

### PR DESCRIPTION
**Pull request by RMartinOscar**
- [x] Add MassPrune
- [x] Add config for webhook prune days
- [x] Add `created: Webhook` to Event blacklist to avoid fork bomb

image

### CODE REVIEW

**Review by Boy132 — APPROVED** [review]

**Comment by notAreYouScared** [comment]
This config option should also be added to the settings page of the panel.

**Comment by RMartinOscar** [comment]
Should I add a global kill for Webhooks ? Like if you have `APP_WEBHOOK_ENABLED=false` then it will clear all webhook from queue and thus not process them ?

File: app/Console/Kernel.php [hunk 1]
```diff
@@ -48,5 +49,9 @@ protected function schedule(Schedule $schedule): void
         if (config('activity.prune_days')) {
             $schedule->command(PruneCommand::class, ['--model' => [ActivityLog::class]])->daily();
         }
+
+        if (config('webhook.prune_days')) {
+            $schedule->command(PruneCommand::class, ['--model' => [WebhookConfiguration::class]])->daily();
```
Comment (lancepioch) [inline app/Console/Kernel.php:54]:
```suggestion
            $schedule->command(PruneCommand::class, ['--model' => [Webhook::class]])->daily();
```

File: app/Models/WebhookConfiguration.php [hunk 2]
```diff
@@ -5,12 +5,50 @@
...
+ * @method static \Illuminate\Database\Eloquent\Builder|WebhookConfiguration whereDescription($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|WebhookConfiguration whereEndpoint($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|WebhookConfiguration whereEvents($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|WebhookConfiguration whereId($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|WebhookConfiguration whereUpdatedAt($value)
+ * @method static \Illuminate\Database\Eloquent\Builder|WebhookConfiguration withTrashed()
+ * @method static \Illuminate\Database\Eloquent\Builder|WebhookConfiguration withoutTrashed()
+ *
+ * @mixin \Eloquent
+ */
```
Comment (lancepioch) [inline app/Models/WebhookConfiguration.php:41]:
```suggestion
```

Don't need this for now.

File: app/Models/WebhookConfiguration.php [hunk 3]
```diff
@@ -118,4 +160,12 @@ public static function discoverCustomEvents(): array
 
         return $events;
     }
+
+    /**
+     * Get the prunable model query.
+     */
```
Comment (lancepioch) [inline app/Models/WebhookConfiguration.php:166]:
```suggestion

```

File: app/Filament/Pages/Settings.php [hunk 4]
```diff
@@ -540,7 +540,21 @@ private function miscSettings(): array
...
+                ->collapsed()
+                ->schema([
+                    TextInput::make('APP_WEBHOOK_PRUNE_DAYS')
+                        ->label('Prune age')
+                        ->required()
+                        ->numeric()
+                        ->minValue(1)
+                        ->maxValue(365)
+                        ->suffix('Days')
+                        ->default(env('APP_WEBHOOK_PRUNE_DAYS', config('webhook.prune_days'))),
```
Comment (lancepioch) [inline app/Filament/Pages/Settings.php:556]:
```suggestion
                        ->default(env('APP_WEBHOOK_PRUNE_DAYS', config('panel.webhooks.prune_days'))),
```

Comment (lancepioch) [reply app/Filament/Pages/Settings.php:556]: Don't make a separate config file for this

**Review by lancepioch — CHANGES_REQUESTED** [review]
Very close

**Review by notAreYouScared — APPROVED** [review]
Tested, LGTM

**Pull request merged**